### ISSUE TITLE
Set option or queries?

### ISSUE THREAD

**Issue by amelieled** [Deducteam/lambdapi#503]
As you can see in the new documentation, there are two subsections named `set`, but I don't understand why some of them are in the first subsection, and the others in the second one.

For me, to extend your own logic, you can define:
 * `symbol`
 * `set` `prefix`
 * `set` `infix`
 * `inductive`
 * `rule`
 * `set` `unif_rule`
or you can include another file with:
 * `require` / `as` 
 * `require`
 * `open`
 But it's not really clear for this command:
  * `set` `builtin` {"0" | "+1" | "T" | "P" | "eq" | "refl" | "eq_ind" | "top" | "bot" | "not" | "or"  | "and" | "imp"}
 
 I think, all of these can be view as axioms except if you can proof them, thanks to tactics:
 * `refine`
 * `assume`
 * `apply`
 * `simpl`
 * `rewrite`
 * `reflexivity`
 * `symmetry`
 * `why3`
 * `focus` (specific one)
 * `fail`     (specific one)

Maybe, we can speak about **set option** when a command doesn't change the current logic, but only the file management (some parameters):
 * `set` `verbose`
 * `set` `debug`
 * `set` `flag` `eta_equality` {on | off}
 * `set` `flag` `print_domains` {on | off}
 * `set` `flag` `print_implicits` {on | off}
 * `set` `flag` `print_meta_type` {on | off}
 * `set` `flag` `print_contexts` {on | off}
 * `set` `flag` `print_domains` {on | off}
 * `set` `prover`
 * `set` `prover_timeout`
 * `set` `declared`
 * `set` `quantifier`

Finally, the queries are only: 
 * `print`
 * `proofterm`
 * `type`
 * `compute`
 * `assert`
 * `assertnot`
and they just give an answer to a question, without changing anything.

When you change the logic, maybe we can replace `set` by `decl` (for declaration).
And probably in the future,  `symbol` / `set` `prefix` / `set` `infix` will be merged, @gabrielhdt ?

To summarize:
  * If you change the logic, you use a **declaration command**
  * If you change file configuration without changing the logic, you setup your file thanks to a **set option**.
  * If you don't change anything, you use a **query**.

**Comment by amelieled** [comment]
This discussion is useful for the PR #492 .

**Comment by fblanqui** [comment]
The set commands of the first section modify the signature state of the system, while the set commands of the second section do not modify the signature state of the system.

**Comment by fblanqui** [comment]
As already mentioned, https://github.com/Deducteam/lambdapi/issues is reserved for real issues. For questions or discussions, please use https://github.com/Deducteam/lambdapi/discussions. I therefore close this "issue".

**Comment by amelieled** [comment]
It's a discussion but a problem for the user too. There are 2 sections `set` and `print` in the documentation.
To solve the second one, maybe we can distinguish `print_goal`, `print_proofterm` and `print` (which will parse to `P_query_print` again).

**Comment by amelieled** [comment]
To solve the first one, it's not clear because the difference is not relevant for the users.

**Comment by fblanqui** [comment]
Fixed in https://github.com/Deducteam/lambdapi/pull/506
